In [1]:
import numpy as np
import pandas as pd

In [2]:
data = pd.read_csv('https://archive.ics.uci.edu/ml/machine-learning-databases/adult/adult.data', names=['age','workclass','fnlwgt','education','education-num','marital-status','occupation','relationship','race','sex','capital-gain','capital-loss','hours-per-week','native-country','salary'])
data.head()

age          workclass  fnlwgt   education  education-num  \
0   39          State-gov   77516   Bachelors             13   
1   50   Self-emp-not-inc   83311   Bachelors             13   
2   38            Private  215646     HS-grad              9   
3   53            Private  234721        11th              7   
4   28            Private  338409   Bachelors             13   

        marital-status          occupation    relationship    race      sex  \
0        Never-married        Adm-clerical   Not-in-family   White     Male   
1   Married-civ-spouse     Exec-managerial         Husband   White     Male   
2             Divorced   Handlers-cleaners   Not-in-family   White     Male   
3   Married-civ-spouse   Handlers-cleaners         Husband   Black     Male   
4   Married-civ-spouse      Prof-specialty            Wife   Black   Female   

   capital-gain  capital-loss  hours-per-week  native-country  salary  
0          2174             0              40   United-States   <=50K  
1             0             0              13   United-States   <=50K  
2             0             0              40   United-States   <=50K  
3             0             0              40   United-States   <=50K  
4             0             0              40            Cuba   <=50K

# 1. How many men and women (sex feature) are represented in this dataset?

In [8]:
data.sex.value_counts()

 Male      21790
 Female    10771
Name: sex, dtype: int64

Hence, 21790 Men and 10771 Women

# 2. What is average age (age feature) of women?

In [6]:
data.query('sex == " Female"').age.mean()

36.85823043357163

In [7]:
data.loc[data.sex==' Female', 'age'].mean()

36.85823043357163

Hence, 36.86

# 3. What is the percentage of German citizens (native-country feature)?

In [12]:
round(float((data['native-country'] == ' Germany').sum()) / data.shape[0]*100, 2)

0.42

Answer: 0.42

# 4. What are the mean and standard deviation of age for those who earn more than 50K  per year (salary feature)?

In [15]:
data.groupby('salary').age.agg(['mean','std']).round(1)

mean   std
salary            
 <=50K  36.8  14.0
 >50K   44.2  10.5

Answer: 44.2 +- 10.5

# 5. What are the mean and standard deviation of age for those who earn less than 50K  per year?

According to the table from (4) it is 36.8 +- 14

# 6. Is it true that people who earn more than 50K have at least high school education (education - Bachelors, Prof-school, Assoc-acdm, Assoc-voc, Masters or Doctorate feature)

In [19]:
data.query('salary==" >50K"').education.unique()

array([' HS-grad', ' Masters', ' Bachelors', ' Some-college',
       ' Assoc-voc', ' Doctorate', ' Prof-school', ' Assoc-acdm',
       ' 7th-8th', ' 12th', ' 10th', ' 11th', ' 9th', ' 5th-6th',
       ' 1st-4th'], dtype=object)

Answer: No

# 7.Find the maximum age of men of Amer-Indian-Eskimo race

In [24]:
data.loc[data.race==' Amer-Indian-Eskimo', 'age'].max()

82

Answer: 82

# 8. Among whom is the proportion of those who earn a lot (>50K) greater: married or single men?

In [37]:
data['marital-status'].str.startswith(' Marri'); #True for Married

data.groupby(data['marital-status'].str.startswith(' Marri')).salary.value_counts()

marital-status  salary
False            <=50K    16039
                 >50K      1105
True             <=50K     8681
                 >50K      6736
Name: salary, dtype: int64

In [40]:
pd.crosstab(data.salary, data['marital-status'].str.startswith(' Marri'), normalize='columns').round(2)

marital-status  False  True 
salary                      
 <=50K           0.94   0.56
 >50K            0.06   0.44

Answer: Among married

# 9. What is the maximum number of hours a person works per week? How many people work such number of hours and what is the percentage of thsose who earn a lot among them?

In [42]:
#max hours
data['hours-per-week'].max()

99

In [44]:
#number of people
data.loc[data['hours-per-week']==data['hours-per-week'].max()].shape[0]

85

In [47]:
data.loc[data['hours-per-week']==data['hours-per-week'].max()].salary.value_counts(normalize=True)

 <=50K    0.705882
 >50K     0.294118
Name: salary, dtype: float64

Answer: 99, 85, 29%

# 10. Count the average time of work for those who earn a little and a lot for each country. What will this be for Japan?

In [50]:
data.groupby(['native-country', 'salary'])['hours-per-week'].mean()

native-country  salary
 ?               <=50K    40.164760
                 >50K     45.547945
 Cambodia        <=50K    41.416667
                 >50K     40.000000
 Canada          <=50K    37.914634
                            ...    
 United-States   >50K     45.505369
 Vietnam         <=50K    37.193548
                 >50K     39.200000
 Yugoslavia      <=50K    41.600000
                 >50K     49.500000
Name: hours-per-week, Length: 82, dtype: float64

In [52]:
data.groupby(['native-country', 'salary'])['hours-per-week'].mean()[' Japan']

salary
 <=50K    41.000000
 >50K     47.958333
Name: hours-per-week, dtype: float64

Answer: 41 and 48